### PR TITLE
Fixed bug #3363: `this` in ajax.url, ajax.data points on element

### PR DESCRIPTION
**Pull request by SilverFire**
For further details visit #3361 

P.S.: Grunt updated some CSS. It seems, that previous contributor didn't run grunt.

### CODE REVIEW

**Comment by SilverFire** [comment]
@kevin-brown ready to merge

File: dist/js/select2.js [hunk 1]
```diff
@@ -3306,11 +3306,11 @@ S2.define('select2/data/ajax',[
     }, this.ajaxOptions);
 
     if (typeof options.url === 'function') {
-      options.url = options.url(params);
+      options.url = options.url(self.$element, params);
     }
 
     if (typeof options.data === 'function') {
-      options.data = options.data(params);
+      options.data = options.data(self.$element, params);
```
Comment (kevin-brown) [inline dist/js/select2.js]: This should be `options.data.call(self.$element, params)` so the `this` is set instead of the first parameter.

**Comment by kevin-brown** [comment]
Right now there's an issue in how you are calling the methods with the `$element` included, where now it is added as the first parameter instead of being used as the `this` reference.

**Comment by SilverFire** [comment]
@kevin-brown :anguished: oh, sorry, that's my fault. Now it seems to be OK?

**Comment by SilverFire** [comment]
:up:

**Comment by kevin-brown** [comment]
Just a note that _I'm not ignoring this_, I'm actually planning on pulling in the changes soon (likely this weekend).

I almost applied this last weekend but ended up burning most of my time on a regression test that just wouldn't fail.

**Comment by SilverFire** [comment]
:up:

**Pull request closed without merging**